apps/ejabberd/src/mongoose_metrics_hooks.erl

```diff
@@ -110,64 +110,75 @@ get_hooks(Host) ->
      [mam_muc_purge_multiple_messages, Host, ?MODULE, mam_muc_purge_multiple_messages, 50]].
 
 
--spec sm_register_connection_hook(tuple(), ejabberd:jid(), term()
+-spec sm_register_connection_hook(map(), tuple(), ejabberd:jid(), term()
                                  ) -> metrics_notify_return().
-sm_register_connection_hook(_,#jid{server = Server}, _) ->
+sm_register_connection_hook(Acc, _,#jid{server = Server}, _) ->
     mongoose_metrics:update(Server, sessionSuccessfulLogins, 1),
-    mongoose_metrics:update(Server, sessionCount, 1).
+    mongoose_metrics:update(Server, sessionCount, 1),
+    Acc.
 
--spec sm_remove_connection_hook(tuple(), ejabberd:jid(),
+-spec sm_remove_connection_hook(map(), tuple(), ejabberd:jid(),
                                 term(), ejabberd_sm:close_reason()
                                ) -> metrics_notify_return().
-sm_remove_connection_hook(_,#jid{server = Server},_, _Reason) ->
+sm_remove_connection_hook(Acc, _,#jid{server = Server},_, _Reason) ->
     mongoose_metrics:update(Server, sessionLogouts, 1),
-    mongoose_metrics:update(Server, sessionCount, -1).
+    mongoose_metrics:update(Server, sessionCount, -1),
+    Acc.
 
--spec auth_failed(binary(), binary()) -> metrics_notify_return().
-auth_failed(_,Server) ->
-    mongoose_metrics:update(Server, sessionAuthFails, 1).
+-spec auth_failed(map(), binary(), binary()) -> metrics_notify_return().
+auth_failed(Acc, _,Server) ->
+    mongoose_metrics:update(Server, sessionAuthFails, 1),
+    Acc.
 
--spec user_send_packet(ejabberd:jid(), tuple(), tuple()
+-spec user_send_packet(map(), ejabberd:jid(), tuple(), tuple()
                       ) -> metrics_notify_return().
-user_send_packet(#jid{server = Server},_,Packet) ->
+user_send_packet(Acc, #jid{server = Server},_,Packet) ->
     mongoose_metrics:update(Server, xmppStanzaSent, 1),
-    user_send_packet_type(Server, Packet).
+    user_send_packet_type(Server, Packet),
+    Acc.
 
 -spec user_send_packet_type(Server :: ejabberd:server(),
-                            Packet :: jlib:xmlel()) -> metrics_notify_return().
+                            Packet :: jlib:xmlel()) -> ok | {error, not_found}.
```

Comment:
THIS IS NOT A HANDLER. I REPEAT THIS IS NOT A HANDLER. It is just a helper function called by handlers, so it is a different return type. Dialyzer say it is correct.